README.md

```diff
@@ -1,9 +1,65 @@
+
+### ID Tokens and PK Tokens
+
+OpenPubkey uses ID Tokens issued by OpenID Providers (OPs) to produce PK Tokens. A PK Token consist of the ID Token and the CIC (Client-Instance Claims) where the CIC contains the user's public key and associated metadata. The ID Token contains a hash of the CIC, so that the values in the CIC including the identities public key are cryptographically bound to the ID Token signed by the OP.
+
+For workload-identity, the hash of the CIC is stored in the ID Token's `aud` claim following the pattern of github actions workload identity. For user-identity, the hash of the CIC is stored in the ID Token's `nonce` claim.
+
+## OpenPubkey FAQ
+
+### What is the difference between Sigstore and OpenPubkey?
+
+OpenPubkey cannot really be compared to Sigstore. Sigstore is an end-to-end artifact signing solution, whereas OpenPubkey only binds public keys to OIDC identities for use as part of a larger signing solution. OpenPubkey is complementary to Sigstore, for example https://github.com/sigstore/fulcio/issues/1056
+
+As stated in the Related Work section of [the OpenPubkey paper](https://eprint.iacr.org/2023/296.pdf):
+
+> “Sigstore [31, 42] is an open source project for signing and verifying software artifacts. Users can sign under their OpenID Connect identity by using the sigstore Fulcio Certificate Authority [43] which uses an immutable log to store a mapping between an OpenID Connect ID Token and a short lived public key enabling parties to attribute signatures to identities. The Fulcio CA (Certificate Authority) is trusted to create this mapping between an ID Token and a public key. Using OpenPubkey this trust can be eliminated as OpenPubkey does not need a trusted party to map ID Tokens to public keys. The Fulcio CA could in turn help OpenPubkey by acting as a public OpenPubkey verifier and OP public key database."
+
+### How does OpenPubkey ensure the nonce claim functions as nonce?
+
+In the user-identity scenario the CIC that contains the user's public key is hashed to the `nonce` claim in the ID Token. As OIDC requires that this field never repeat, OpenPubkey includes a random value, rz, in the CIC. Thus the hash of the CIC is always different and random. This maintains the required properties needed by the `nonce` claim in OIDC.
+
+```
+rz = crypto.random()
+CIC = {'rz': crypto.random(), 'upk': <publickey>, 'alg': 'EC256'}
+IDToken.nonce = SHA3(CIC)
+```
+
+### Does OpenPubkey present a privacy leak?
+
+The PK Tokens used by OpenPubkey contain the claims from the OIDC ID Tokens of the signer, so making them public, necessarily makes those claims public too. This may include elements of the signer’s identity such as the signer’s name or email addresses. It is up to users of OpenPubkey as to whether or not PK Tokens are made public.
```

Comment:
```suggestion
The PK Tokens used by OpenPubkey contain the claims from the OIDC ID Tokens of the signer, so making them public necessarily makes those claims public too. This may include elements of the signer’s identity such as the signer’s name or email addresses. It is up to users of OpenPubkey as to whether or not PK Tokens are made public.
```

(remove comma after "so making them public")